[ToggleButton] Update styles for Material v2

mbrookes opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

- No drop-shadow
- Outside border
- Square buttons
- Vertical separator between buttons
- Lighter colors

<img width="1024" alt="image" src="https://user-images.githubusercontent.com/357702/51571605-a1ceeb00-1e9a-11e9-885f-09fe35b52c84.png">

oliviertassinari commented:
This is almost like a button group:
- https://blueprintjs.com/docs/#core/components/button-group
- https://ant.design/components/button/#components-button-demo-button-group
- https://getbootstrap.com/docs/4.2/components/button-group/

One pattern emerges out, I think that the toggle button group height should match the button height.

We might have an issue with the wrapping logic:

![capture d ecran 2019-01-23 a 00 34 58](https://user-images.githubusercontent.com/3165635/51572735-d5b00d80-1ea6-11e9-8711-720499849ebb.png)

mbrookes commented:
This should not have been so difficult, but got there in the end! 😅 

What do you think about having a `size` / `variant` / `dense` prop which reproduces the compact vertical sizing of the v1 toggle buttons, but in flat v2 style?

mbrookes commented:
>  What do you think about having a `size` / `variant` / `dense` prop

Never mind,  it's easy enough to override the height.

mbrookes commented:
Okay, I'm done. 😆

mbrookes commented:
Okay, I'm done. 😆

joshwooding commented:
When I select multiple buttons the width changes?

![toggle_buttons](https://user-images.githubusercontent.com/12938082/52527850-5f454500-2cc7-11e9-9e62-44b45adaa3aa.gif)

mbrookes commented:
> When I select multiple buttons the width changes?

Damn, looks like I squashed prematurely! It was fine before I refactored it...

Let me figure out when I broke it.

oliviertassinari commented:
In the future, I think that it would be great to introduce a button group component.

mbrookes commented:
👌
